gui/default/syncthing/core/syncthingController.js

```diff
@@ -1686,17 +1744,234 @@ angular.module('syncthing.core')
+                        .then(function (resp) {
+                            // Set excludeCredentials in frontend instead of backend so we can be consistent with UI state
+                            resp.data.publicKey.excludeCredentials = (($scope.webauthn.state || {}).credentials || []).map(function (cred) {
+                                return { type: "public-key", id: cred.id };
+                            });
+                            return webauthnJSON.create(resp.data);
+                        })
+                        .then(function (pkc) {
+                            return $http.post(urlbase + '/webauthn/register-finish', pkc);
+                        })
+                        .then(function (resp) {
+                            if (!$scope.webauthn.state) {
+                                $scope.webauthn.state = {};
+                            }
+                            if (!$scope.webauthn.state.credentials) {
+                                $scope.webauthn.state.credentials = [];
+                            }
+                            $scope.webauthn.state.credentials.push(resp.data);
+                        })
+                        .catch(function (e) {
+                            if (e instanceof DOMException && e.name === "InvalidStateError") {
+                                $scope.webauthn.errors.alreadyRegistered = true;
+                            } else if (e instanceof DOMException && e.name === "AbortError") {
+                                $scope.webauthn.errors.aborted = true;
+                            } else if (e instanceof DOMException && e.name === "NotAllowedError") {
+                                $scope.webauthn.errors.notAllowed = true;
+                            } else if (e instanceof DOMException && e.name === "SecurityError") {
+                                $scope.webauthn.errors.securityError = true;
+                            } else {
+                                $scope.webauthn.errors.registrationFailed = true;
+                                console.log('Credential creation failed:', e);
+                            }
+                        });
+                };
+
+                $scope.deleteWebauthnCredential = function (cred) {
+                    $scope.webauthn.state.credentials = $scope.webauthn.state.credentials.filter(function (cr) {
+                        return cr.id !== cred.id;
+                    });
+                };
+
+                $scope.saveWebauthnState = function () {
+                    return $http.post(
+                        urlbase + '/webauthn/state',
+                        $scope.webauthn.state,
+                        { headers: { 'Content-Type': 'application/json' } }
+                    ).then(function() {
+                        $scope.webauthn.editingCredentialIds = {};
+                    }).finally(function () {
+                        console.log('saveWebauthnState', $scope.webauthn.state);
```

Comment:
Debugging leftover?

Reply:
Fixed in 86e9193685e9c523c6c6d163c24e3def17afaa20 (part of https://github.com/emlun/syncthing/pull/7).